dcicutils/misc_utils.py

```diff
@@ -984,17 +986,109 @@ def str_to_bool(x: Optional[str]) -> Optional[bool]:
         raise ValueError(f"An argument to str_or_bool must be a string or None: {x!r}")
 
 
-def to_integer(value: str, fallback: Optional[Any] = None) -> Optional[Any]:
+def to_integer(value: str, fallback: Optional[Any] = None, strict: bool = False) -> Optional[Any]:
     try:
         return int(value)
     except Exception:
-        try:
-            return int(float(value))
-        except Exception:
-            pass
+        if strict is not True:
+            try:
+                return int(float(value))
+            except Exception:
+                pass
     return fallback
 
 
+_MULTIPLIER_K = 1000
+_MULTIPLIER_M = 1000 * _MULTIPLIER_K
+_MULTIPLIER_G = 1000 * _MULTIPLIER_M
+_MULTIPLIER_T = 1000 * _MULTIPLIER_G
+
+_MULTIPLIER_SUFFIXES = {
+    "K": _MULTIPLIER_K,
+    "KB": _MULTIPLIER_K,
+    "M": _MULTIPLIER_M,
+    "MB": _MULTIPLIER_M,
+    "G": _MULTIPLIER_G,
+    "GB": _MULTIPLIER_G,
+    "T": _MULTIPLIER_T,
+    "TB": _MULTIPLIER_T
+}
+
+
+def to_number(value: str,
+              allow_commas: bool = False,
+              allow_multiplier_suffix: bool = False,
+              allow_float: bool = False,
+              fallback: Optional[Union[int, float]] = None) -> Optional[Union[int, float]]:
+
+    """
+    Converts the give string value to an int, or possibly float if allow_float is True.
```

Comment:
`possibly float if allow_float is True`

seems to be the _default_ behavior based on your test?

Reply:
I reworked this, and tightened up tests.